nestjs-counter/apps/server/src/exchange-rates/exchange-rates.spec.ts

```diff
@@ -0,0 +1,31 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { ExchangeRatesService } from './exchange-rates.service';
+import { exchangeRatesProviders } from './exchange-rates.providers';
+
+describe('ExchangeRatesService', () => {
+  let app: TestingModule;
+  const rates = { AUD: 1.5 };
+
+  beforeAll(async () => {
+    const handleMock = {
+      query: jest.fn(() => rates),
+    };
```

Comment:
Why mock here? I'd rather show how to test a real app instead of relying on mocks

Reply:
2 reasons:

1) I had a fair bit of trouble getting the test env to work with Jest and Nest.
2) Strict unit testing (stub everything, only test the code you're looking at) seems to be more idiomatic for Nest.

I don't hold very strong opinions here. Personally, I'd prefer to test a real app in this case. But I'd also like to keep this sample as idiomatic as reasonably possible for Nest. So I'd love to hear any thoughts or feedback.